iconv behaviour on x86_64-musl

## Expected Behaviour

When building and running checks off of the `20230517` release on x86_64 it should pass without issue

## Actual Behaviour
An error is thrown (output from the the build on voidlinux)
```
=> neomutt-20230517_1: running do_check ...
ccache cc -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe  -D_FILE_OFFSET_BITS=64 -DLARGEFILE_SOURCE=1  -ffile-prefix-map=/builddir/neomutt-20230517/neomutt-20230517=. -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/include/lua5.4 -I/usr/include -I/usr/include -I/usr/include -DNCURSES_WIDECHAR -I/usr/include -I/usr/include -I/usr/include -I. -I. -Wall  -I./test -MT test/attach/mutt_actx_add_fp.o -MD -MP -MF test/attach/mutt_actx_add_fp.Tpo -c -o test/attach/mutt_actx_add_fp.o test/attach/mutt_actx_add_fp.c
ccache cc -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe  -D_FILE_OFFSET_BITS=64 -DLARGEFILE_SOURCE=1  -ffile-prefix-map=/builddir/neomutt-20230517/neomutt-20230517=. -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/include/lua5.4 -I/usr/include -I/usr/include -I/usr/include -DNCURSES_WIDECHAR -I/usr/include -I/usr/include -I/usr/include -I. -I. -Wall  -I./test -MT test/attach/mutt_actx_free.o -MD -MP -MF test/attach/mutt_actx_free.Tpo -c -o test/attach/mutt_actx_free.o test/attach/mutt_actx_free.c
ccache cc -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe  -D_FILE_OFFSET_BITS=64 -DLARGEFILE_SOURCE=1  -ffile-prefix-map=/builddir/neomutt-20230517/neomutt-20230517=. -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/include/lua5.4 -I/usr/include -I/usr/include -I/usr/include -DNCURSES_WIDECHAR -I/usr/include -I/usr/include -I/usr/include -I. -I. -Wall  -I./test -MT test/attach/mutt_actx_new.o -MD -MP -MF test/attach/mutt_actx_new.Tpo -c -o test/attach/mutt_actx_new.o test/attach/mutt_actx_new.c
ccache cc -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe  -D_FILE_OFFSET_BITS=64 -DLARGEFILE_SOURCE=1  -ffile-prefix-map=/builddir/neomutt-20230517/neomutt-20230517=. -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/include/lua5.4 -I/usr/include -I/usr/include -I/usr/include -DNCURSES_WIDECHAR -I/usr/include -I/usr/include -I/usr/include -I. -I. -Wall  -I./test -MT test/base64/mutt_b64_buffer_decode.o -MD -MP -MF test/base64/mutt_b64_buffer_decode.Tpo -c -o test/base64/mutt_b64_buffer_decode.o test/base64/mutt_b64_buffer_decode.c
test/attach/mutt_actx_add_fp.c: In function 'test_mutt_actx_add_fp':
test/attach/mutt_actx_add_fp.c:34:5: error: variable 'fp' has initializer but incomplete type
   34 |     FILE fp = { 0 };
      |     ^~~~
test/attach/mutt_actx_add_fp.c:34:17: warning: excess elements in struct initializer
   34 |     FILE fp = { 0 };
      |                 ^
test/attach/mutt_actx_add_fp.c:34:17: note: (near initialization for 'fp')
test/attach/mutt_actx_add_fp.c:34:10: error: storage size of 'fp' isn't known
   34 |     FILE fp = { 0 };
      |          ^~
test/attach/mutt_actx_add_fp.c:34:10: warning: unused variable 'fp' [-Wunused-variable]
make: *** [Makefile:692: test/attach/mutt_actx_add_fp.o] Error 1
make: *** Waiting for unfinished jobs....
=> ERROR: neomutt-20230517_1: do_check: '${make_check_pre} ${make_cmd} ${makejobs} ${make_check_args} ${make_check_target}' exited with 2
=> ERROR:   in do_check() at common/build-style/configure.sh:32
```

## Steps to Reproduce

```sh
# Clone the branch with the changes needed to update neomutt on void
git clone https://github.com/AnInternetTroll/void-packages.git
cd void-packages
# In this commit the changes are reproducable
git checkout fd734a5dc406ed1ec3909ddd6c7c657f573cc93e

# Bootstrap x86_64-musl
./xbps-src -m masterdir-musl binary-bootstrap x86_64-musl
# Build and test package
./xbps-src -m masterdir-musl -Q pkg neomutt
```

## How often does this happen?

- Always
  + On x86_64-musl only. No other architecture supported by voidlinux has this issue

## When did it start to happen?

- When I upgraded
  Which version did you use to use?
  + 20220429

## NeoMutt Version

`20230517`

(no "neomutt -v" available)

## Extra Info

<!-- This info might help us narrow down the source of the problem. -->

* Operating System and its version
  + Void Linux freshly installed on May 17th 2023

--- a/test/convert/mutt_convert_file_to.c
+++ b/test/convert/mutt_convert_file_to.c
@@ -87,8 +87,7 @@ void test_mutt_convert_file_to(void)
     struct Content info = initial_info;
 
     size_t rc = mutt_convert_file_to(fp, "iso-8859-2", tocodes, &tocode, &info);
-    TEST_CHECK(rc == -1);
-    TEST_MSG("Check failed: %d == -1", rc);
+    TEST_CHECK(rc != sizeof(data) - 1);
 
     slist_free(&tocodes);
     fclose(fp);
@@ -105,8 +104,7 @@ void test_mutt_convert_file_to(void)
     struct Content info = initial_info;
 
     size_t rc = mutt_convert_file_to(fp, "iso-8859-2", tocodes, &tocode, &info);
-    TEST_CHECK(rc == -1);
-    TEST_MSG("Check failed: %d == -1", rc);
+    TEST_CHECK(rc != sizeof(data) -1);
 
     slist_free(&tocodes);
     fclose(fp);

